tests/unit/introspection.tcl

```diff
@@ -419,12 +419,25 @@ start_server {tags {"introspection"}} {
     # known keywords. Might be a good idea to avoid adding tests here.
 }
 
-start_server {tags {"introspection"} overrides {protected-configs {yes} debug-commands-disabled {yes}}} {
-    test {cannot modify protect configs} {
+start_server {tags {"introspection external:skip"} overrides {enable-protected-configs {no} enable-debug-command {no}}} {
+    test {cannot modify protected configuration - no} {
         assert_error "ERR*protected*" {r config set dir somedir}
-    }
+        assert_error "ERR*DEBUG command not allowed*" {r DEBUG OBJECT x}
+    } {} {needs:debug}
+}
 
-    test {debug commands disabled} {
-        assert_error "ERR*DEBUG commands not allowed*" {r DEBUG OBJECT x}
+start_server {config "minimal.conf" tags {"introspection external:skip"} overrides {protected-mode {no} enable-protected-configs {local} enable-debug-command {local}}} {
+    test {cannot modify protected configuration - local} {
+        assert_no_match "ERR*protected*" {r config set dir somedir}
+        assert_no_match "ERR*protected*" {r DEBUG OBJECT x}
+
+        # Get a non-loopback address of this instance for this test.
+        set myaddr [get_nonloopback_addr]
+        if {$myaddr != "" && ![string match {127.*} $myaddr]} {
+            # Non-loopback client should fail
+            set r2 [get_nonloopback_client]
+            assert_error "ERR*protected*" {$r2 config set dir somedir}
+            assert_error "ERR*DEBUG command not allowed*" {$r2 DEBUG OBJECT x}
```

Comment:
do we have a test that verifies that local connections succeeds wen the configs are set to `local`?

Reply:
yes the first two lines (`r` is a local connection)
        assert_no_match "ERR*protected*" {r config set dir somedir}
        assert_no_match "ERR*protected*" {r DEBUG OBJECT x}

Reply:
ohh, i was confused by the `assert_no_match` (didn't notice the `not`).
let's assert for what we expect rather than what we don't expect.
or if we don't know what we expect, just call the command with no `catch` and add a comment that we just verify it doesn't error.